### PR TITLE
Better axis fonts for spectrum viewers

### PR DESCRIPTION
**Pull request by pllim**
### Description

This pull request is to make spectrum viewer axes more readable.

Right now:

* Axis label font stuff is hardcoded in `app.vue`.
* Tick label font stuff is not applied until data is loaded.

### Cubeviz

⬇️ Lab + Dark mode + 27" monitor

cubeviz_darkmode_27in_monitor

⬇️ Notebook + Light mode + 13" monitor (the viewer didn't fit on the whole screen)

cubeviz_lightmode_13in_monitor

⬇️ Standalone (voila) + Dark mode + 13" monitor (Y-axis label cut off on spectrum viewer)

cubeviz_voila_darkmode_13in_monitor

### Mosviz

⬇️ Lab + Dark mode + 27" monitor

specviz2d_darkmode_27in_monitor

⬇️ Notebook + Light mode + 13" monitor (the viewer didn't fit on the whole screen, Y-axis label got cut off a little)

mosviz_lightmode_13in_monitor

⬇️ Standalone (voila) + Light mode (cannot get dark mode to work and it is out of scope anyway and Mosviz parser is not really that robust in standalone mode) + 13" monitor (Y-axis label cut off on spectrum viewer, both spectrum viewers really not useful)

mosviz_voila_lightmode_13in_monitor

### Specviz

⬇️ Lab + Dark mode + 27" monitor

specviz_darkmode_27in_monitor

⬇️ Notebook + Light mode + 13" monitor (the viewer didn't fit on the whole screen)

specviz_lightmode_13in_monitor

⬇️ Standalone (voila) + Dark mode + 13" monitor

specviz_voila_darkmode_13in_monitor

### Specviz2d

⬇️ Lab + Dark mode + 27" monitor

mosviz_darkmode_27in_monitor

⬇️ Notebook + Light mode + 13" monitor (the viewer didn't fit on the whole screen)

specviz2d_lightmode_13in_monitor

⬇️ Standalone (voila) + Dark mode + 13" monitor (Y-axis label is going to cut off for longer flux units)

specviz2d_voila_darkmode_13in_monitor

### TODO

- [x] "bigger and darker fonts"
- [x] Update screenshots
- [x] What about small screen?
- [x] What about dark mode?

### Change log entry

- [x] Is a change log needed? If yes, is it added to `CHANGES.rst`? If you want to avoid merge conflicts,
  list the proposed change log here for review and add to `CHANGES.rst` before merge. If no, maintainer
  should add a `no-changelog-entry-needed` label.

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [x] Are two approvals required? Branch protection rule does not check for the second approval. If a second approval is not necessary, please apply the `trivial` label.
- [x] Do the proposed changes actually accomplish desired goals? Also manually run the affected example notebooks, if necessary.
- [x] Do the proposed changes follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [x] Are tests added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [x] Are docs added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [x] Did the CI pass? If not, are the failures related?
- [x] Is a milestone set? Set this to bugfix milestone if this is a bug fix and needs to be released ASAP; otherwise, set this to the next major release milestone.
- [ ] After merge, any internal documentations need updating (e.g., JIRA, Innerspace)? [🐱](https://jira.stsci.edu/browse/JDAT-3707)

### CODE REVIEW

**Comment by kecnry** [comment]
not that it never could before, but increasing the label size does result in the label not fitting in the viewer height (especially for long units in cubeviz where the viewer is fairly height-challenged):

image

**Comment by pllim** [comment]
I updated the screenshots in the original posts. They are from these combos:

1. WSL2 (Debian) + Chrome on Windows 10 + Lab (dark mode) + 27" monitor
2. Windows 11 + Chrome + Notebook + 13" monitor
3. Windows 11 + Chrome + Voila (standalone, some dark, some light) + 13" monitor

File: jdaviz/configs/cubeviz/plugins/viewers.py
Comment (kecnry) [inline jdaviz/configs/cubeviz/plugins/viewers.py:40]: I wonder if this part at least deserves a change log entry?

File: jdaviz/configs/mosviz/plugins/viewers.py
Comment (kecnry) [inline jdaviz/configs/mosviz/plugins/viewers.py:248]: can we safely remove this by any chance now that we override `num_ticks` elsewhere?

**Review by kecnry — COMMENTED** [review]
I still think we might eventually want to make this customizable (out of scope for now) and worry about losing space for the actual data, but we can always iterate further and this does add a more unified feel to text size across the app for the default case.

Just have a few small comments questions and can then approve.  Thanks!

File: jdaviz/configs/mosviz/plugins/viewers.py
Comment (pllim) [reply jdaviz/configs/mosviz/plugins/viewers.py:248]: But you set it to 8 elsewhere. I find 5 better for Mosviz due to how crowded it is.

File: jdaviz/configs/cubeviz/plugins/viewers.py
Comment (pllim) [reply jdaviz/configs/cubeviz/plugins/viewers.py:40]: Change log added.

File: jdaviz/configs/mosviz/plugins/viewers.py
Comment (kecnry) [reply jdaviz/configs/mosviz/plugins/viewers.py:248]: then maybe we should just set it to 5 everywhere?  I'm just trying to avoid specific overrides as much as possible that are going to be a headache to maintain in the future.... but if you think this should be 5 and the other should be 8, then this makes sense

**Review by kecnry — APPROVED** [review]

File: jdaviz/configs/mosviz/plugins/viewers.py
Comment (pllim) [reply jdaviz/configs/mosviz/plugins/viewers.py:248]: I do think Mosviz needs less tick marks than the others, yes.

**Comment by pllim** [comment]
Can we count Patrick's Slack comment as second approval?

**Comment by kecnry** [comment]
@bmorris3 was still going to review, iirc

**Review by PatrickOgle — APPROVED** [review]
Spectral viewer axes label sizes look good.

File: jdaviz/configs/cubeviz/plugins/viewers.py
Comment (bmorris3) [reply jdaviz/configs/cubeviz/plugins/viewers.py:40]: Just in case anyone wants it the old way, could we add a note in the Cubeviz docs in case people want to see it the old way (with axes)? The user could call:
```python
for viewer_name in ['flux-viewer', 'uncert-viewer']:
    cubeviz.app.get_viewer(viewer_name).state.show_axes = True
```

Comment (bmorris3) [reply jdaviz/configs/cubeviz/plugins/viewers.py:40]: Should we also update the Cubeviz docs screenshots and animations?

**Review by bmorris3 — COMMENTED** [review]
Some quick thoughts below, happy to approve after your response. 🎉

File: jdaviz/configs/cubeviz/plugins/viewers.py
Comment (kecnry) [reply jdaviz/configs/cubeviz/plugins/viewers.py:40]: (and also a note that it can be toggled via the UI in plot options)

Comment (pllim) [reply jdaviz/configs/cubeviz/plugins/viewers.py:40]: > Should we also update the Cubeviz docs screenshots and animations?

That should be its own ticket. I think way more than that is outdated by now. 😬 

> Just in case anyone wants it the old way

I think we agreed that `.app` isn't public API, so I would rather not advertise that. But I can mention the Plot Options.

Comment (pllim) [reply jdaviz/configs/cubeviz/plugins/viewers.py:40]: Added a commit to update Plot Option docs. I think only Cubeviz is affected.

**Comment by pllim** [comment]
I responded to @bmorris3 and this PR already has 2 approvals, so I am merging when CI passes (devdeps failed with unrelated error that should be fixed upstream already, so I restarted it).

**Pull request merged**
